docs/source/llms/transformers/guide/index.rst

```diff
@@ -648,4 +678,156 @@ This input format requires that both the bitrate has been set prior to conversio
+
+Example Usage
+^^^^^^^^^^^^^
+
+Here is the example of using ``save_pretrained`` argument for logging a model
+
+.. code-block:: python
+
+    import transformers
+
+    pipeline = transformers.pipeline(
+        task="text-generation", model="databricks/dolly-v2-7b", torch_dtype="torch.float16"
+    )
+
+    with mlflow.start_run():
+        mlflow.transformers.log_model(
+            transformers_model=pipeline,
+            artifact_path="dolly",
+            save_pretrained=False,
+        )
+
+MLflow will not save the copy of Dolly-v2-7B model weight, instead logs the following metadata as a reference to the HuggingFace Hub model. This will save storage space about 15GB and reduce the logging latency significantly as well.
+```
+source_model_name: "databricks/dolly-v2-7b"
+source_model_revision: "d632f0c8b75b1ae5b26b250d25bfba4e99cb7c6f"
+```
+
+.. _caveats-of-save-pretrained:
+
+Caveats of Reference-Only Models
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+While the ``save_pretrained`` argument is useful for saving storage space and reducing logging latency, it has the following caveats to be aware of:
+
+* **Change in Model Unavailability**: If you are using a model from other users' repository, themodel may be deleted or become private in the HuggingFace Hub. In such case, MLflow cannot load the model back. For production use cases, it is recommended to save the copy model weight to the artifact store once for your production model.
+
+* **HuggingFace Hub Access**: The model downloading from the HuggingFace Hub might be slow or unstable due to the network condition or the HuggingFace Hub service condition. MLflow doesn't provide any retry mechanism or robust error handling for the model downloading.
+
+* **Limited Databricks Integration**: If you are using Databricks, be aware that the model saved with `save_pretrained=False` cannot be registered to the legacy `Workspace Model Registry <https://docs.databricks.com/en/machine-learning/manage-model-lifecycle/workspace-model-registry.html>`_. If you want to register the reference-only Transformer model, please use `Unity Catalog <https://docs.databricks.com/en/machine-learning/manage-model-lifecycle/index.html>`_ instead, or download the model weight in advance using :py:func:`mlflow.transformers.persist_pretrained_model()` API as described in the next section.
+
+.. _persist-pretrained-guide:
+
+Persist the Model Weight to the Existing Reference-Only Model
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+If you want to update the reference-only model to the one contains the model weight, you can use :py:func:`mlflow.transformers.persist_pretrained_model()` API. This API will download the model weight from the HuggingFace Hub, save it to the artifact location, and update the metadata of the given reference-only model. After this operation, the model will be equivalent to the one saved with `save_pretrained=True` and be ready for the production use.
```

Comment:
```suggestion
If you want to update the reference-only model to an instance that contains the model weight, you can use the :py:func:`mlflow.transformers.persist_pretrained_model()` API. This API will download the model weight from the HuggingFace Hub, save it to the artifact location, and update the metadata of the given reference-only model. After this operation, the model will be equivalent to the one saved with `save_pretrained=True` and be ready for the production use.
```
Can we add a callout to be very clear to users that they don't have to re-log a new run in order to update the artifact with the model weights with some nice bold text to REALLY warn them about this?

Reply:
Great idea! Adding a tip box and stress the fact that the update doesn't require re-logging.